main.js

```diff
@@ -0,0 +1,213 @@
+    const uuid = Date.now().toString();
+    context[uuid] = fn;
+    const res = context[uuid](...rest, ...args);
+    delete context[uuid];
+    return res;
+  };
+}
+const cardOfDog = {
+  breedOfDog: "",
+  nameOfDog: "",
+  fullInfo() {
+    return `I love my ${this.breedOfDog} his named is ${this.nameOfDog}.`;
+  },
+};
+
+function getFullProfile(breedOfDog, nameOfDog) {
+  this.breedOfDog = breedOfDog;
+  this.nameOfDog = nameOfDog;
+  return this.fullInfo();
+}
+
+bind(getFullProfile, cardOfDog, "Amstaff", "Oscar")();
+
+// // 3.
+
+// /*  Cтворити функцію, котра приймає 2 параметри - об'єкти.
+// Функція повинна перевіряти чи ці 2 обʼєкти
+// абсолютно ідентичні та повертає результат у зрозумілому форматі */
+
+function deepCompare() {
+  let i, l, leftChain, rightChain;
+  function compareTwoObjects(x, y) {
+    let p;
+    if (
+      isNaN(x) &&
+      isNaN(y) &&
+      typeof x === "number" &&
+      typeof y === "number"
+    ) {
+      return true;
+    }
+    if (x === y) {
+      return true;
+    }
+    if (
+      (typeof x === "function" && typeof y === "function") ||
+      (x instanceof Date && y instanceof Date) ||
+      (x instanceof RegExp && y instanceof RegExp) ||
+      (x instanceof String && y instanceof String) ||
+      (x instanceof Number && y instanceof Number)
+    ) {
+      return x.toString() === y.toString();
+    }
+    if (!(x instanceof Object && y instanceof Object)) {
+      return false;
+    }
+    if (x.isPrototypeOf(y) || y.isPrototypeOf(x)) {
+      return false;
+    }
+    if (x.constructor !== y.constructor) {
+      return false;
+    }
+    if (x.prototype !== y.prototype) {
+      return false;
+    }
+    if (leftChain.indexOf(x) > -1 || rightChain.indexOf(y) > -1) {
+      return false;
+    }
+    for (p in y) {
+      if (y.hasOwnProperty(p) !== x.hasOwnProperty(p)) {
+        return false;
+      } else if (typeof y[p] !== typeof x[p]) {
+        return false;
+      }
+    }
+    for (p in x) {
+      if (y.hasOwnProperty(p) !== x.hasOwnProperty(p)) {
+        return false;
+      } else if (typeof y[p] !== typeof x[p]) {
+        return false;
+      }
+      switch (typeof x[p]) {
+        case "object":
+        case "function":
+          leftChain.push(x);
+          rightChain.push(y);
+          if (!compareTwoObjects(x[p], y[p])) {
+            return false;
+          }
+          leftChain.pop();
+          rightChain.pop();
+          break;
+        default:
+          if (x[p] !== y[p]) {
+            return false;
+          }
+          break;
+      }
+    }
+    return true;
+  }
+  if (arguments.length < 1) {
+    return true;
+  }
+  for (i = 1, l = arguments.length; i < l; i++) {
+    leftChain = [];
+    rightChain = [];
+    if (!compareTwoObjects(arguments[0], arguments[i])) {
+      return false;
+    }
+  }
+  return true;
+}
+
+deepCompare(1, null); // false
+deepCompare(NaN, NaN); // true
+deepCompare("smoke", "smoke"); // true
+
```

Comment:
Nice solution, but we only need to compare 2 objects. No need to compare string, NaN, null etc.

Reply:
Carefully with task description